### PR TITLE
chore: decouple post processor from gapicgen

### PR DESCRIPTION
**Pull request by codyoss**
- This will make it easier to onboard new clients by configuring
  only two files: OwlBot.yaml and config.yaml
- Remove all deps on gapicgen. This meant inlining more logic into
  the post processor
- Split out logic into domain files
- Add idea of a post processor config that reads file dynamically
  ensuring we can onboard new clients without re-deploying post
  processor
- Use a heuristic to determine a release-level

### CODE REVIEW

**Comment by codyoss** [comment]
Note once review is done I will add the updated post-processor owlbot revision

File: internal/postprocessor/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+postprocessor
```
Comment (quartzmo) [inline internal/postprocessor/.gitignore:1]: Why is this needed?

Comment (codyoss) [reply internal/postprocessor/.gitignore:1]: If you run `go build` in this directory it produces a binary called postprocessor, this makes sure that never gets uploaded to github

File: internal/postprocessor/config.yaml [hunk 2]
```diff
@@ -124,3 +124,435 @@ modules:
   - webrisk
   - websecurityscanner
   - workflows
+
+service-configs:
```
Comment (quartzmo) [inline internal/postprocessor/config.yaml:128]: Would it be possible to sort these alphabetically?

Comment (codyoss) [reply internal/postprocessor/config.yaml:128]: They are, by input directory

File: internal/postprocessor/mainifest.go [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+	for scanner.Scan() && lineCnt < 50 {
+		line := scanner.Text()
+		if strings.Contains(line, betaIndicator) {
+			return "beta", nil
+		}
+	}
+	return "ga", nil
+}
+
+var manualEntries = []ManifestEntry{
```
Comment (quartzmo) [inline internal/postprocessor/mainifest.go:146]: I know these are very stable, but just have to ask: Should this be extracted to a config file too?

File: internal/postprocessor/config.yaml [hunk 4]
```diff
@@ -555,4 +559,111 @@ service-configs:
...
+ - input-directory: google/cloud/translate/v3
+   service-config: translate_v3.yaml
+   import-path: cloud.google.com/go/translate/apiv3
+ - input-directory: google/cloud/vision/v1
+   service-config: vision_v1.yaml
+   import-path: cloud.google.com/go/vision/v2/apiv1
+ - input-directory: google/cloud/vision/v1p1beta1
+   service-config: vision_v1p1beta1.yaml
+   import-path: cloud.google.com/go/vision/v2/apiv1p1beta1
+ - input-directory: google/cloud/workstations/v1beta
```
Comment (quartzmo) [inline internal/postprocessor/config.yaml:664]: Unsorted, move `workstations` down?

File: internal/postprocessor/config.yaml (hunk 2 above)
Comment (quartzmo) [reply internal/postprocessor/config.yaml:128]: Regarding sort order, see comment below on `workstations`.

Comment (quartzmo) [reply internal/postprocessor/config.yaml:128]: Can modules now be derived from service-configs?

Comment (codyoss) [reply internal/postprocessor/config.yaml:128]: They can not, a module can have N clients at different levels and modules don't have to be directly off the root.

File: internal/postprocessor/config.yaml (hunk 4 above)
Comment (codyoss) [reply internal/postprocessor/config.yaml:771]: It is sorted, maybe this was in an older commit

File: internal/postprocessor/mainifest.go (hunk 3 above)
Comment (codyoss) [reply internal/postprocessor/mainifest.go]: I like the idea, done

File: internal/postprocessor/config.yaml [hunk 5]
```diff
@@ -0,0 +1,776 @@
...
+ - input-directory: google/spanner/admin/instance/v1
+   service-config: spanner_admin_instance.yaml
+ - input-directory: google/spanner/v1
+   service-config: spanner.yaml
+ - input-directory: google/storage/v2
+   service-config: storage_v2.yaml
+ - input-directory: google/storagetransfer/v1
+   service-config: storagetransfer_v1.yaml
+   # All entries below are unmigrated clients. We need to store extra
+   # info for them since they don't have an entry in owlbot.yaml yet.
```
Comment (quartzmo) [inline internal/postprocessor/config.yaml:671]: Do some of the unmigrated clients keys shadow entries above this line? For example, 'google/devtools/cloudtrace/v1'?

Comment (codyoss) [reply internal/postprocessor/config.yaml:601]: Oops you are right, I generated the initial set before I realized non-migrated clients needs more info. Updated
